src/plugins/embeddable/public/lib/embeddables/embeddable.tsx

```diff
@@ -131,6 +132,28 @@ export abstract class Embeddable<
     return this.output;
   }
 
+  public async getExplicitInputIsEqual(lastInput: Partial<TEmbeddableInput>): Promise<boolean> {
+    const currentInput = this.getExplicitInput();
+    return (
+      genericEmbeddableInputIsEqual(lastInput, currentInput) &&
+      fastIsEqual(omitGenericEmbeddableInput(lastInput), omitGenericEmbeddableInput(currentInput))
+    );
+  }
+
+  public getExplicitInput() {
+    if (this.getRoot().isContainer) {
+      return (
+        ((this.getRoot() as IContainer).getInput().panels?.[this.id]
```

Comment:
I think we have a type guard for isContainer 
`getIsContainer`

Reply:
Oh good call, I will implement that!